lib/strategy.ts

```diff
@@ -0,0 +1,93 @@
+import { HeaderIndexes, DatasetMapping } from "./models";
+
+
+export interface SortingStrategy {
+    doSort(data: string[], datasetMapping?: DatasetMapping, headerIndexes?: HeaderIndexes, getType?: (row: string[], headerIndexes: HeaderIndexes) => string): string[];
+    toggleSortOrder(): void;
+}
+
+export enum SortState {
+    Off = 0,
+    Ascending = 1,
+    Descending = 2
+}
+
+abstract class SortingStrategyBase implements SortingStrategy {
+    public sortState: SortState = SortState.Ascending;
+
+    public toggleSortOrder(): void {
+        this.sortState = (this.sortState + 1) % 3;
+    }
+
+    abstract doSort(data: string[], datasetMapping: DatasetMapping, headerIndexes: HeaderIndexes, getType?: (row: string[], headerIndexes: HeaderIndexes) => string): string[];
```

Comment:
Suggestion (non-blocking): getType doesn't need to be a parameter, you can import the function in this file instead